File: app/javascript/channels/room_channel.js
import consumer from "./consumer"
const chatChannel =consumer.subscriptions.create("RoomChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    // return alert(data['message']);
    return $('#messages').append(data['message']);
  },

  speak: function(data) {
    // return this.perform('speak');
    // let messages = JSON.perse(message);
    let insert_html = '';
    insert_html += '<div class="message">';
    insert_html += '  <p>' + data + '</p>';
    insert_html += '</div>';

    return $('#messages').append(insert_html);
  }
});


$(document).on('keypress', '[data-behavior~=room_speaker]', function(event) {
  if (event.keyCode === 13) {
    chatChannel.speak(event.target.value);
    event.target.value = '';
    return event.preventDefault();
  }
});

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "room_channel"
    stream_from "room_channel_#{params['room']}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    # ActionCable.server.broadcast 'room_channel', message: data['message']
    # Message.create! user_id: 2, room_id: 1, content: data['message']
    # logger.debug("data:" << data.inspect)
    Message.create! user_id: current_user.id, room_id: params['room'], content: data['message']
  end
end
